premium/usage.go

```diff
@@ -405,14 +405,16 @@ func (u *BatchUpdater) getTeamNameByTokenType(tokenType auth.TokenType) (string,
 			return "", fmt.Errorf("expected to find exactly one team for API key, found %d", len(resp.JSON200.Items))
 		}
 		return resp.JSON200.Items[0].Name, nil
-	case auth.SyncRunAPIKey, auth.SyncTestConnectionAPIKey:
+	default:
```

Comment:
Could this have consequences in the future where it would allow other (potential) user-accessible keys to be used and spoof the team name, at least in older plugins?

Reply:
If you run plugins/CLI locally you can always spoof the environment variables. The backend still enforces API keys can only access what they are permitted to do.

If this code runs in managed syncs we can control which environment variables to pass and decide not to pass `_CQ_TEAM_NAME` for other tokens

Reply:
CLI: Backend still enforces api keys but the team name could be different, and with an "unhandled" new type of user key (and an older plugin version, which doesn't explicitly support that key) this code would allow a different team name to be reported.

Reply:
> this code would allow a different team name to be reported.

What would be the impact of that? API keys are tied to a team to they only have a single team so if the team is spoofed auth will fail. For user tokens the impact would be to report usage not on the team specified in the config, which they can modify regardless?
If we want to enforce usage reporting limitations it shouldn't be in the client code

Reply:
yeah the keys are still tied to a team so it wouldn't really do anything, ok good :)